MediaBrowser.Common/Json/Converters/JsonNullableGuidConverter.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Globalization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace MediaBrowser.Common.Json.Converters
+{
+    /// <summary>
+    /// Converts a GUID object or value to/from JSON.
+    /// </summary>
+    public class JsonNullableGuidConverter : JsonConverter<Guid?>
+    {
+        /// <inheritdoc />
+        public override Guid? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var guidStr = reader.GetString();
+            return guidStr == null ? null : new Guid(guidStr);
+        }
+
+        /// <inheritdoc />
+        public override void Write(Utf8JsonWriter writer, Guid? value, JsonSerializerOptions options)
+        {
+            if (value == null || value == Guid.Empty)
```

Comment:
Can't think of why anyone would use guid.empty - but are we saying guid.empty will be converted to null if saves and re-loaded?

Reply:
In 10.6 all Guid.Empty were converted to `null` on serialization; we can't do that because the api client needs to know if a parameter is nullable.

Imo this is more of a stopgap until we can go through and determine which Guid fields can be made nullable